app/controllers/posts_controller.rb

```diff
@@ -6,5 +6,31 @@ def index
 
   def show
     @post = Post.find(params[:id])
+    @comment = Comment.new
+    @like = Like.new
+  end
+
+  def new
+    @post = Post.new
+    @params = params
+  end
+
+  def create
+    @post = Post.new(posts_params)
+    if @post.save
+      redirect_to user_posts_path(id: @post.id, author_id: @post.author_id,
+                                  comments_counter: @post.comments_counter,
+                                  likes_counter: @post.likes_counter)
```

Comment:
- [ ] A post should be created on behalf of the currently logged in user (current_user) of the app

Reply:
Hi @mikethreels , I appreciate the review. However, I am not satisfied with the reason for the required changes because the code is working according to the requirements.